### ISSUE TITLE
Atlas migrations don't work with sqlite fts5 virtual tables

### ISSUE THREAD

**Issue by eric** [ent/ent#3928]
Our schema has a [sqlite FTS5 virtual table](https://www.sqlite.org/fts5.html) (created manually) and we found that when we try to use Ent to create another automatic migration that it fails with the following error:

```
2024/02/01 17:18:23 failed generating migration file: clean schemas after migration replaying: drop "imdb_fts_data" table: no such table: imdb_fts_data
```

The issue is that the virtual table `imdb_fts` creates the table `imdb_fts_data` and when the `imdb_fts` virtual table is dropped, it drops the underlying `imdb_fts_data` table.

When the cleanup code runs:

https://github.com/ent/ent/blob/1bcb2fa9ccd7e53040decd1c5d64c4a1ee0d1d28/dialect/sql/schema/atlas.go#L202-L217

it fails when it can't find `imdb_fts_data`.

To work around this, I was able to just use the `IF EXISTS` indicator when deleting fancybits/ent@1aab698171328421578509ffc103c7c135a45536 (I did run into an issue with Atlas support for `IF EXISTS` with sqlite that I have a fix for: ariga/atlas#2499).

I don't know what the best solution is to this problem in a general sense, but wanted to raise the issue.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.21  |
| Ent         | 0.12.5   |
| Database    | sqlite3   |
| Driver      | github.com/mattn/go-sqlite3        |

**Comment by giautm** [comment]
Hi @eric again, maybe the fix should plant in Atlas too. Because it should know the relationship between two tables and ignore `imdb_fts_data` from inspection.

**Comment by eric** [comment]
If Ent/Atlas doesn't understand FTS5 (we've created the virtual table outside of Ent/Atlas using SQL migrations) I'm not sure how it would know to ignore those tables... but if you have a way, that's great!

**Comment by giautm** [comment]
That's true, atlas doesn't support FTS5 yet. There is open issue for it https://github.com/ariga/atlas/issues/2118 For this case, I think the fix on atlas side can help you workaround currently?

**Comment by eric** [comment]
No, I still had to fork Ent for this change fancybits/ent@1aab698171328421578509ffc103c7c135a45536 to make it work. It would be nice if I didn't have to fork Ent for this but I couldn't find anywhere I could hook into to make that happen.

**Comment by giautm** [comment]
Can you make a PR? I think it's safe to merge.

**Comment by eric** [comment]
I've submitted #3931
